src/pages/Apps/actions.js

```diff
@@ -35,19 +35,20 @@ export const useFetchAppList = ({ api }) => {
 
 const appCache = {};
 
-export const useFetchApp = ({ api, appId }) => {
+export const useFetchApp = ({ api, appId, location }) => {
   const [app, setApp] = useState(appCache[appId] || {});
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
+  const refresh = location?.state?.app;
```

Comment:
This is a useful hack which forces any action (anywhere in the app) to use `location.push({ state: { app: Date.now() }})` to trigger an app refetch.